serveradmin/servershell/views.py

```diff
@@ -197,8 +198,21 @@ def get_results(request):
 
 @login_required
 def inspect(request):
-    server = Query({'object_id': request.GET['object_id']}, None).get()
-    return _edit(request, server, template='inspect')
+    if request.method != 'GET':
+        return HttpResponseNotAllowed(['GET'])
+
+    if 'hostname' in request.GET:
```

Comment:
If `hostname` AND `object_id` are specified in the GET request, wouldn't it make more sense to favor the `object_id` over the `hostname` as it would be more precise?

Reply:
I will swap them since that makes more sense but as pointed out in the description of the PR this endpoint is really just used by the Servershell which always uses the object_id (pro grammatically) and this addition for the hostname is just a convenience feature for some power users who really should know what they are doing.

On the very long term I aim for getting rid of the speciality of the hostname attribute anyways. We already store plenty of information about projects, networks etc. where we "abuse" the hostname as an identifier. So this will probably go away or change in the future is should just be seen as a intermediate code for power users.

Reply:
Regards of the long term it makes even more sense to switch it, so it won't be a behavior change when you fill in both query parameters.